lyse/figure_manager.py

```diff
@@ -140,4 +140,4 @@ def install():
     figuremanager = FigureManager()
     matplotlib.pyplot.figure = figuremanager
     matplotlib.pyplot.close = figuremanager.close
-    matplotlib.pyplot.show = figuremanager.show
+    matplotlib.pyplot.show = lambda: figuremanager.show
```

Comment:
@dihm Doesn’t this need to be `lambda: figuremanager.show()`?

Reply:
Interesting. Apparently both ways work. I'll admit my understanding of `lambda` isn't very deep, but now that you point it out I suspect you are correct in the strictest sense. Does it work as-is because it's just passing through a passed-through function handle?

I'm not sure it is worth changing, but I will if you think it is.

Reply:
Definitely needs to be `lambda: figuremanager.show()`.

`lambda: figuremanager.show` does nothing, it defines a function that returns a function without calling it.  It appears to 'work' because explicitly calling `show()` is not necessary when running an analysis routine within lyse - all plots got shown regardless.

But when running an analysis routine script from outside of lyse, the figure doesn't get shown.

Reply:
This feels really dumb, but how do you actually run analysis scripts outside of the lyse GUI? I tried running some tests that way and couldn't get past `Run` instantiation failing on the call to `self._frame_to_path(outer_frame)`. I suspect it's something simple I'm not doing right, but I couldn't find it in the docs so figured I'd ask then add to the docs for future me.

Reply:
@dihm it's possible it's broken. Looks like it's meant to raise a KeyError. What is it raising instead?

Reply:
I'm getting this traceback running my dumb script in the same folder as the shot via `python script.py`

```shell
Traceback (most recent call last):
  File ".\LyseDecoratorEdgeCases_PR102.py", line 4, in <module>
    shot = Run('LyseDecoratorTestShot.h5')
  File "c:\users\naqsl\src\labscript-suite\lyse\lyse\__init__.py", line 313, in __init__
    outer_path = self._frame_to_path(outer_frame)
  File "c:\users\naqsl\src\labscript-suite\lyse\lyse\__init__.py", line 330, in _frame_to_path
    path = frame.f_globals['__file__']
AttributeError: 'NoneType' object has no attribute 'f_globals'
```

Reply:
Pretty sure that is meant to work. I think it's also meant to detect the filename too and use it for the HDF5 group.

The except clause that catches KeyError should trigger of you didn't have a file and were manually using lyse from an interactive terminal (in which case you have no filename and need to manually tell it the HDF5 group to use when saving results)

Reply:
Looks like the functionality for `Run()` objects figuring out the group name when the script is run outside of lyse has been broken at some point. The code in `Run.__init__()` is popping out on the stack one level at a time, explicitly waiting to see when it gets to the `analysis_subprocess` module as an indicator that it has gone too far - but when running outside of lyse, `analysis_subprocess` is not used, so it never finds it.

This can be done way more simply. Some time ago we made the user's script more convincingly appear to be the `__main__` module, so we can find it that way, which will work when running outside of lyse as well. And no stack traversal. https://github.com/labscript-suite/lyse/pull/106 should fix.